lua/rest-nvim/utils/init.lua

```diff
@@ -43,32 +43,70 @@ M.read_file = function(file)
   return lines
 end
 
--- read_env_file Reads the environment variables found in the env_file option
--- (defualt: .env) specified in configuration and returns a table with the
--- variables
-M.read_env_file = function()
+-- get_variables Reads the environment variables found in the env_file option
+-- (defualt: .env) specified in configuration or from the files being read
+-- with variables beginning with @ and returns a table with the variables
+M.get_variables = function()
   local variables = {}
-  local env_file = "/" .. (config.get("env_file") or ".env")
-
-  -- Directories to search for env files
-  local env_file_paths = {
-    -- current working directory
-    vim.fn.getcwd() .. env_file,
-    -- directory of the currently opened file
-    vim.fn.expand("%:p:h") .. env_file,
-  }
-
-  -- If there's an env file in the current working dir
-  for _, env_file_path in ipairs(env_file_paths) do
-    if M.file_exists(env_file_path) then
-      for line in io.lines(env_file_path) do
-        local vars = M.split(line, "=", 1)
-        variables[vars[1]] = vars[2]
-      end
-    end
-  end
 
-  return variables
+	-- If there is a line at the beginning with @ first
+	if vim.fn.search('^@', 'cn') > 0 then
+		-- Read all lines of the file
+		local lines = vim.api.nvim_buf_get_lines(0, 0, -1, true);
+
+		-- For each line
+		for _, line in pairs(lines) do
+			-- Get all that starts with @
+			if line:match("^@[%w!@#$%^&*-_+?~]+%s*=%s*[^=]+") then
```

Comment:
Since we are already using a string.match here. Code from line 60-67 can just be compressed into a single line using capture groups in lua regex.

```lua
local name, val = line:match("^@([%w!@#$%^&*-_+?~]+)%s*=%s*([^=]+)$")

if name then
  variables[name] = val
end
```